### DISCUSSION TITLE
Problem when installing Directus on top of an existing SQL Server database

### DISCUSSION THREAD

**Discussion by kavalcante** [directus/directus#5751] [Archive]
I did a fresh install of Directus on top of an existing SQL Server database. After the installation when running the project, I received a lot of errors that the primary key was not defined. Here's some examples:
![image](https://user-images.githubusercontent.com/13931503/118721708-950bf900-b801-11eb-838f-03f746f8a952.png)

After running the project I also received an error of a select, just like the image below:
![image](https://user-images.githubusercontent.com/13931503/118721783-ad7c1380-b801-11eb-8fa7-2fba51664162.png)

Does anyone know's how to solve this?

**Comment by rijkvanzanten** [comment]
How did you install? Did the database migrations run correctly when installing?

What types do your primary keys have in those tables? Directus scans through the constraint names for the fields to find the primary key constraint. What constraint names do your primary keys have?

**Reply by kavalcante** [reply]
> How did you install?

I installed with the CLI, like the documentation said.

> Did the database migrations run correctly when installing?

I think so. During the installing does not appeared any error message.

> What types do your primary keys have in those tables?

They're integers.

> What constraint names do your primary keys have?

The name of the column from each table can be different. For example, in table `home_destaques_portal_emresumo`, the name of the primary key is `codItem`.

**Reply by rijkvanzanten** [reply]
@kavalcante Would you be able to share a database dump so I can have a look? Looks like Directus isn't able to accurately extract what column is the primary key, but it's a little hard to tell/reproduce why 🤔  (you can truncate the actual rows in the tables, or any columns that aren't the PK if there's any sensitive data in there of course. You can also send it directly to me on rijk@directus.io if you'd rather not share it on this public forum 👍🏻 )
